README.md

```diff
@@ -18,17 +18,28 @@ yarn add -D \
     prettier
 ```
 
-In your `eslint.config.cjs`:
+In your `eslint.config.mjs`:
 
 ```js
-const foxglove = require("@foxglove/eslint-plugin");
-const tseslint = require("typescript-eslint");
+import foxglove from "@foxglove/eslint-plugin";
+import tseslint from "typescript-eslint";
 
-module.exports = tseslint.config(
+export default tseslint.config(
   ...foxglove.configs.base,
   ...foxglove.configs.react,
   ...foxglove.configs.jest,
-  ...foxglove.configs.typescript
+  ...foxglove.configs.typescript.map((config) => ({
+    ...config,
+    files: ["**/*.ts"],
```

Comment:
@jtbandes any reason why this is not part of the plugin export? it was a surprising amount of boilerplate

Reply:
What happened if you omitted this?

Maybe there is a better default to be had. Though in some ways I think it is nice if the exported configs don’t assume anything about the user’s directory structure or file extension preferences (for example what if you only wanted to include files in `src`?)

Reply:
What happened if you omitted this?

Maybe there is a better default to be had. Though in some ways I think it is nice if the exported configs don’t assume anything about the user’s directory structure or file extension preferences (for example what if you only wanted to include files in `src`?)

Reply:
I'm not sure about errors related to js vs ts specifically.

Years ago when I set up some of the original yaml rules I remember it being a problem, for example if there were any stray js files in the repo (e.g. config files) they would sometimes get caught up in unnecessary typescript lint rules.

Reason for making this PR is that I tried to update dbmate using the existing example from this readme, and ran into errors. In hindsight I think it was probably due to the lack of `parserOptions` (added [https://github.com/amacneil/dbmate/blob/c6e23f928b0322294eb952cc45960621aec1507d/typescript/eslint.config.mjs#L13-L20](here)), not the scoping to `*.ts` files specifically.

But when I hit the error, I had a look at some other foxglove repos you had migrated, e.g. [https://github.com/foxglove/template-typescript/blob/main/eslint.config.cjs](template-typescript/eslint.config.cjs), and saw that you had a lot more code there than what was in the readme of this repo. So this PR is basically me copying the template-typescript stuff into this readme.

Feel free to edit/merge/reject, I don't have any strong opinions on it, just documenting what worked for dbmate.

Reply:
Here's an example where I didn't need any additional settings to make things work: https://github.com/foxglove/message-definition/blob/main/eslint.config.cjs

That repo is much simpler so maybe you only run into problems with sufficiently "interesting" repos.

I think for now I'll just close this – but will revisit if anyone else reports difficulty with using the plugin.